inst/include/cpp11/protect.hpp

```diff
@@ -42,6 +221,7 @@ SEXP unwind_protect(Fun&& code) {
 
   std::jmp_buf jmpbuf;
   if (setjmp(jmpbuf)) {
+    *preserved.should_unwind_protect_ = TRUE;
```

Comment:
This one is important, otherwise we don't correctly set the flag after there is an error.